contrib/debian/rules

```diff
@@ -12,10 +12,16 @@ override_dh_auto_clean:
 	if [ -f Makefile ]; then $(MAKE) distclean; fi
 	rm -rf Makefile.in aclocal.m4 configure src/Makefile.in src/bitcoin-config.h.in src/build-aux src/qt/Makefile.in src/qt/test/Makefile.in src/test/Makefile.in
 
+ifeq ($(shell dpkg-vendor --derives-from Ubuntu && echo yes),yes)
```

Comment:
```
QT=$(shell dpkg-vendor --derives-from Ubuntu && echo qt4 || echo qt5)
```

or better yet (to enable Qt5 on Ubuntu 17.10+ which is GNOME instead of Unity):

```
QT=$(shell { [ $(lsb_release -s -i) = Ubuntu ] && { [ $(lsb_release -s -r | cut -d. -f1) -lt 17 ] || [ $(lsb_release -s -r) = 17.04 ]; } } && echo qt4 || echo qt5)
```